feature/experimental/system/gnmi/benchmarking/ate_tests/internal/setup/setup.go

```diff
@@ -0,0 +1,296 @@
+	lspBit.SetBit = ygot.Bool(false)
+	isisTimers := globalISIS.GetOrCreateTimers()
+	isisTimers.LspLifetimeInterval = ygot.Uint16(600)
+	spfTimers := isisTimers.GetOrCreateSpf()
+	spfTimers.SpfHoldInterval = ygot.Uint64(5000)
+	spfTimers.SpfFirstInterval = ygot.Uint64(600)
+
+	isisLevel1 := isis.GetOrCreateLevel(1)
+	isisLevel1.Enabled = ygot.Bool(false)
+
+	isisLevel2 := isis.GetOrCreateLevel(2)
+	isisLevel2.Enabled = ygot.Bool(true)
+	isisLevel2.MetricStyle = oc.Isis_MetricStyle_WIDE_METRIC
+
+	isisLevel2Auth := isisLevel2.GetOrCreateAuthentication()
+	isisLevel2Auth.Enabled = ygot.Bool(true)
+	isisLevel2Auth.AuthPassword = ygot.String(authPassword)
+	isisLevel2Auth.AuthMode = oc.IsisTypes_AUTH_MODE_MD5
+	isisLevel2Auth.AuthType = oc.KeychainTypes_AUTH_TYPE_SIMPLE_KEY
+
+	for _, dp := range dut.Ports() {
+		// Interfaces config.
+		i := d.GetOrCreateInterface(dp.Name())
+		i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+		if *deviations.InterfaceEnabled {
+			i.Enabled = ygot.Bool(true)
+		}
+		i.Description = ygot.String("from oc")
+		i.Name = ygot.String(dp.Name())
+
+		s := i.GetOrCreateSubinterface(0)
+		s4 := s.GetOrCreateIpv4()
+		if *deviations.InterfaceEnabled {
+			s4.Enabled = ygot.Bool(true)
+		}
+		a4 := s4.GetOrCreateAddress(DUTIPList[dp.ID()].String())
+		a4.PrefixLength = ygot.Uint8(plenIPv4)
+
+		// BGP neighbor configs.
+		nv4 := bgp.GetOrCreateNeighbor(ATEIPList[dp.ID()].String())
+		nv4.PeerGroup = ygot.String(PeerGrpName)
+		if dp.ID() == "port1" {
+			nv4.PeerAs = ygot.Uint32(ateAs2)
+		} else {
+			nv4.PeerAs = ygot.Uint32(ATEAs)
+		}
+		nv4.Enabled = ygot.Bool(true)
+
+		// ISIS configs.
+		isisIntf := isis.GetOrCreateInterface(dp.Name())
+		isisIntf.Enabled = ygot.Bool(true)
+		isisIntf.HelloPadding = oc.Isis_HelloPaddingType_ADAPTIVE
+		isisIntf.CircuitType = oc.Isis_CircuitType_POINT_TO_POINT
+
+		isisIntfAuth := isisIntf.GetOrCreateAuthentication()
+		isisIntfAuth.Enabled = ygot.Bool(true)
+		isisIntfAuth.AuthPassword = ygot.String(authPassword)
+		isisIntfAuth.AuthMode = oc.IsisTypes_AUTH_MODE_MD5
+		isisIntfAuth.AuthType = oc.KeychainTypes_AUTH_TYPE_SIMPLE_KEY
+
+		isisIntfLevel := isisIntf.GetOrCreateLevel(2)
+		isisIntfLevel.Enabled = ygot.Bool(true)
+
+		isisIntfLevelTimers := isisIntfLevel.GetOrCreateTimers()
+		isisIntfLevelTimers.HelloInterval = ygot.Uint32(1)
+		isisIntfLevelTimers.HelloMultiplier = ygot.Uint8(5)
+
+		isisIntfLevelAfi := isisIntfLevel.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST)
+		isisIntfLevelAfi.Metric = ygot.Uint32(200)
+		isisIntfLevelAfi.Enabled = ygot.Bool(true)
+	}
+	p := gnmi.OC()
+	fptest.LogQuery(t, "DUT", p.Config(), d)
+
+	return d
+}
+
+// ConfigureATE configures ATE ports with IPv4, ISIS and BGP peers.
+func ConfigureATE(t *testing.T, ate *ondatra.ATEDevice) {
+	topo := ate.Topology().New()
+
+	for _, dp := range ate.Ports() {
+		ISISMetricArray = append(ISISMetricArray, ISISMetric)
```

Comment:
I don't see the need to export ISISMetricArray or ISISSetBitArray. They repeat the same values for the number of ports, but that may just be simpler to let the caller do. (Again, hard to evaluate because there are no users of this field in the PR).

Reply:
Resolved by removing all the content which is not needed by this PR.